### PR TITLE
feat(scaling): Tell Windows ES is high dpi aware using a manifest file

### LINKED ISSUE #7133: Endless Sky does not override Windows DPI scaling by default
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

If you have a higher-resolution monitor on Windows (something above 1080p, definitely with 4k), you'll likely be using DPI scaling to prevent icons and text from being minuscule on screen. However, if Windows does not detect that an application has any DPI scaling awareness, it will simply take that window and blow it up to a larger size with a bilinear filter, giving a blurry image. Endless Sky is one of those applications, despite us supporting higher DPIs with the Main Zoom View setting.

### Steps to Reproduce

1. Open Windows Display settings
2. Set the "Change the size of of text, apps, and item" (DPI scaling) to a value above 100%.
3. Open Endless Sky, observe how the game will be blurrier than at 100% scaling.

### Expected Behavior

The game will run at the same DPI scale, regardless of what is set in Windows.

### Screenshots

In this screenshot, Windows DPI scaling is set to 150%. On the left, we have an Endless Sky instance with no scaling by default, which can't even fit the 100 main zoom value in this specific window. On the right, we have an instance which overrides Windows' scaling, letting it shoot up to 120 main zoom, looking much crisper.

image

### Link to save file

_No response_

### Operating System

Windows 10 Pro

### Game Source

Built from source

### Game Version

commit SHA e1615b1

### Additional Information

If you use DPI scaling on Windows, you can manually disable it for Endless Sky by finding your .exe, right clicking, opening Properties > Compatibility > Change high DPI settings > High DPI scaling override, and selecting the checkbox there.
image

### PR DESCRIPTION
**Pull request by quyykk**
**Bugfix:** This PR fixes #7133 

## Fix Details

This PR embeds an application manifest on Windows, specifying that the game is DPI aware and doesn't need any scaling done by Windows itself.

## Testing Done

Set my DPI to 175%. Starting the game with this PR results in a not zoomed-in game.

### CODE REVIEW

File: endless-sky.exe.manifest [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  />
+  <asmv3:application>
+    <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">
+      <dpiAware>true/pm</dpiAware>
+    </asmv3:windowsSettings>
+    <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">
+      <dpiAwareness>PerMonitorV2, permonitor, unaware</dpiAwareness>
+    </asmv3:windowsSettings>
+  </asmv3:application>
+</assembly>
```
Comment (Terin) [inline endless-sky.exe.manifest:15]:
```suggestion
  <assemblyIdentity type="win32"
                    name="io.github.endless-sky"
                    version="0.9.14.0"
  />
  <asmv3:application>
    <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">
      <dpiAware>true/pm</dpiAware>
    </asmv3:windowsSettings>
    <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">
      <dpiAwareness>PerMonitorV2, permonitor, unaware</dpiAwareness>
    </asmv3:windowsSettings>
  </asmv3:application>
</assembly>

```
Get that CI working! Still builds just fine on my end with this.

File: endless-sky.exe.manifest [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0" xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+  <assemblyIdentity type="win32"
+                    name="io.github.endless-sky"
+                    version="0.9.14.0"
```
Comment (samrocketman) [inline endless-sky.exe.manifest:5]: credits.txt has 0.9.15-alpha should this match?

**Review by petervdmeer — APPROVED** [review]
Looks quite straightforward

**Comment by quyykk** [comment]
Once we update the SDL version we ship with Windows we won't need this anymore and we can use SDL to enable high dpi.

But in the meantime, this'll do.

**Comment by quyykk** [comment]
@tehhowch Would you be ok with this or would you prefer we update the SDL2 we ship on Windows (and add the 3 lines required)?

**Comment by petervdmeer** [comment]
> Once we update the SDL version we ship with Windows we won't need this anymore and we can use SDL to enable high dpi.
> 
> But in the meantime, this'll do.

Good argument to merge this now, at least to bridge the gap until we update the SDL version.

**Pull request merged**